src/controllers/setFavorites.ts

```diff
@@ -0,0 +1,61 @@
+import { FastifyReply, FastifyRequest} from 'fastify';
+import { StatusCodes } from 'http-status-codes';
+import { IStorage } from '../storage';
+import auth from '../middlewares/auth';
+import '../types';
+
+
+export default {
+    init: (storage: IStorage) => {
+        return {
+            schema: {
+                body: {
+                    type: 'object',
+                    required: ['id'],
+                    properties: {
+                        id: {
+                            type: 'string'
+                        },
+                        favorites: {
+                            type: 'boolean'
+                        }
+                    }
+                }
+            },
+            preHandler: auth,
+            handler: async function ( request: FastifyRequest, response: FastifyReply ) {
+                if ( !request.user ) {
```

Comment:
these kind of things is better to separate in the middleware function and attach on the route level

Reply:
It's a good idea, reworked.